### ISSUE TITLE
Groq fails with: Failed to decode JSON output

### ISSUE THREAD

**Issue by glindberg2000** [letta-ai/letta#1309]
**Describe the bug**
When using the Groq endpoint with MemGPT, I encounter an error that prevents successful conversation completion. The error arises during the parsing of JSON output from the local language model (LLM). The error message indicates a failure to decode valid MemGPT JSON from the LLM output.

**Please describe your setup**
- [x] How did you install memgpt?
  - Pip install (`pip install pymemgpt`) and Docker.
- [x] Describe your setup
  - OS: Linux
  - Running `memgpt` in both pip-installed and Docker environments.
- [x] MemGPT Config
  
DB: Postgres from local and memgpt Docker image, tried sqlite3
embedding_endpoint = https://embeddings.memgpt.ai
embedding_model = BAAI/bge-large-en-v1.5
Model: llama3 7B and 70B

**Screenshots**
![image](https://github.com/cpacker/MemGPT/assets/148168828/68431b69-2638-4bb9-bf4d-1ba1f7f40770)

**Additional context**
The error occurs consistently after a few (sometimes longer) rounds of conversation using the Groq endpoint. The conversation starts successfully but fails during the parsing of JSON output from the LLM especially once it starts using function calls or getting near the context limit but the token limit errors (429) are separate from this JSON parsing error.

---

**Local LLM details**

- [x] The exact model you're trying to use: Groq endpoint
- [x] The local LLM backend you are using: Local configuration
- [x] Your hardware for the local LLM backend: Local computer with Linux operating system

**Comment by sarahwooders** [comment]
I believe this is a model issue - we are tying to work with better errors for model failures.

**Comment by cpacker** [comment]
Could potentially be fixed by #1257

**Comment by bear0330** [comment]
This issue seems like caused by this line in memgpt\local_llm\llm_chat_completion_wrappers\chatml.py
```
       assistant_prefix = self.assistant_prefix_extra_first_message if first_message else self.assistant_prefix_extra
        if assistant_prefix and raw_llm_output[: len(assistant_prefix)] != assistant_prefix:
            # print(f"adding prefix back to llm, raw_llm_output=\n{raw_llm_output}")
            raw_llm_output = assistant_prefix + raw_llm_output
            # print(f"->\n{raw_llm_output}")
```
in my test, here assistant_prefix is 
```
'\n{\n  "function":'
```
and raw_llm_output becomes a malform json:
```
'\n{\n  "function":I agree with the frontend development plans. Using Next.js or Create React App will provide a solid foundation for building a scalable and maintainable frontend application. I also like the idea of using Tailwind CSS for its utility-first approach and customizability. For state management, I think Redux would be a good choice, as it provides a robust way to manage global state and props drilling. Let\'s implement the component-driven architecture and set up Storybook for component development and testing. How do you want to approach the implementation of the payment processing service using Stripe\'s API?'
```
thanks.
